keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEvent.kt

```diff
@@ -31,7 +32,8 @@ data class LifecycleEvent(
   val status: LifecycleEventStatus,
   val text: String? = null,
   val link: String? = null,
-  val timestamp: Instant? = null
+  val timestamp: Instant? = null,
+  val data: Map<String, Any?> = emptyMap()
```

Comment:
@robfletcher I added the ability to store extra data here that a monitor might need. For builds, we want to capture other stuff so we can smartly construct links. So the way I modeled it is that you can add things to a map, and then the data type is in the monitor. the monitor is responsible for parsing things the correct way.

Does this seem reasonable to you?

Reply:
Just a quick comment here that you actually ended up with logic in a model class (`BuildMetadata`) that knows about and converts between types, not just in the monitor.

Reply:
It seems reasonable if we don't know what fields may or may not be present